cclib/io/moldenwriter.py

```diff
@@ -215,15 +184,38 @@ def generate_repr(self):
         index = -1
         molden_lines.extend(self._coords_from_ccdata(index))
 
-        # Either both [GTO] and [MO] should be present or none of them.
-        if hasattr(self.ccdata, 'gbasis') and hasattr(self.ccdata, 'mocoeffs')\
-                and hasattr(self.ccdata, 'moenergies'):
+        mocoeffs=None
+        mooccs=None
+        moenergies = None
+        if not self.naturalorbitals and hasattr(self.ccdata, 'moenergies') and hasattr(self.ccdata, 'mocoeffs'):
+            moenergies = self.ccdata.moenergies
+            mocoeffs = self.ccdata.mocoeffs
+            mooccs = numpy.zeros((len(self.ccdata.homos),len(moenergies[0])))
+            occval = 2 // len(self.ccdata.homos)
+            for i in range(len(self.ccdata.homos)):
+                mooccs[i][0:self.ccdata.homos[i]+1] = occval
+        elif self.naturalorbitals and hasattr(self.ccdata, 'nooccnos') and hasattr(self.ccdata, "nocoeffs"):
+            moenergies = numpy.array([self.ccdata.nooccnos])
+            mocoeffs = numpy.array([self.ccdata.nocoeffs])
+            mooccs = numpy.array([self.ccdata.nooccnos])
+        elif self.naturalorbitals and hasattr(self.ccdata, 'nsooccnos') and hasattr(self.ccdata, "nsocoeffs"):
+            moenergies = self.ccdata.nsooccnos
+            mocoeffs = self.ccdata.nsocoeffs
+            mooccs = self.ccdata.nsooccnos
+
+        mosyms = None
+        if hasattr(self.ccdata, 'mosyms') and not self.naturalorbitals:
+            mosyms = self.ccdata.mosyms
+        else:
+            mosyms = numpy.full_like(moenergies, 'A', dtype=str)
 
+        # Either both [GTO] and [MO] should be present or none of them.
```

Comment:
I know this comment is carried over, but how so?  You could print the basis but not MO coefficients and vice versa, then just dump what you have to MOLDEN.  (If this comment is wrong, then we can do a separate PR.)

Reply:
I think it would make sense to have GTO present without MO. But without the GTO present, MOs don't have a lot of meaning?

Reply:
From a usability point, sure.  But there's no technical limitation that prevents MOs from appearing without the GTO section.